LSPS5/README.md

```diff
@@ -0,0 +1,512 @@
+    allowed to return this error, and the client MUST either
+    set up with a protocol that the LSP supports, or not have
+    webhook notifications.
+    An LSP MUST support HTTPS at the minimum.
+
+### Webhook Call
+
+The LSP SHOULD contact the given webhook URI, if:
+
+* The client has registered one via `lsps5.set_webhook`
+* *and* the client currently does not have a BOLT8 tunnel
+  with the LSP (i.e. it is currently not connected to the
+  LSP), except if the client has just successfully registered
+  a webhook.
+* *and* one of the following conditions is true:
+  * The client has just successfully registered a webhook
+    (i.e. successful call of `lsps5.set_webhook`); the LSP
+    SHOULD do this even if the client is currently connected
+    to the LSP.
+  * The LSP learns of an incoming payment to the client
+    (whether an incoming HTLC, or other future mechanism
+    for incoming payments).
+    * For HTLCs (or in the future, PTLCs), the LSP SHOULD hold
+      onto the incoming H/PTLC for a reasonable time that
+      allows the client to awaken, but short enough that the
+      rest of the published network is not unduly impacted.
+  * An HTLC or other time-bound contract, in either direction,
+    in a channel with the client, is about to time out.
+  * The LSP wants to take back some of its liquidity towards
+    the client.
+  * The LSP wants to change feerates for any channels towards
+    the client, or feerates for any auto-management mechanism
+    the client has set up with the LSP.
+  * The LSP wants to withdraw all services from the client
+    and close all channels with the client.
+
+The LSP generates a [JSON-RPC 2.0 Notification Object][] for
+the webhook notification it wants to make.
+
+[JSON-RPC 2.0 Notification Object]: https://www.jsonrpc.org/specification#notification
+
+```JSON
+{
+  "jsonrpc": "2.0",
+  "method": "lsps5.webhook_registered",
+  "params": { }
+}
+```
+
+`jsonrpc` is a required string field, which MUST always be
+the JSON string `"2.0"`.
+
+`method` is the required name of the notification, and MUST
+be one of the webhook notification methods listed later in
+this specification.
+The `method` specifies exactly what the LSP wants to notify
+to the client as a push notification.
+
+`params` is a required by-name parameter object for the
+notification method.
+The parameters may be empty, if the method allows the
+parameters to be empty.
+
+The LSP creates a timestamp of when the notification is
+created, then signs both the timestamp and the above body.
+
+The signature is generated as described in [LND `signmessage`
+#specinatweet][], with the LSP using its node ID for signing.
+
+[LND `signmessage` #specinatweet]: https://web.archive.org/web/20191010011846/https://twitter.com/rusty_twit/status/1182102005914800128
+
+The message to be signed is the following JSON string template:
+
+    "LSPS5: DO NOT SIGN THIS MESSAGE MANUALLY: LSP: At ${timestamp} I notify ${body}"
```

Comment:
All our `signmessage` things need to do this because `signmessage` is a user-facing API.

The second `LSP:` is because this arises from the LSP and not the client. The numeric identifier is at the start of the string.

Reply:
Is current text fine or ....?